nsxt/resource_nsxt_policy_service.go

```diff
@@ -433,9 +440,28 @@ func resourceNsxtPolicyServiceCreate(d *schema.ResourceData, m interface{}) erro
 
 	// Create the resource using PATCH
 	log.Printf("[INFO] Creating service with ID %s", id)
-	err = client.Patch(id, obj)
-	if err != nil {
-		return handleCreateError("Service", id, err)
+
+	if isPolicyGlobalManager(m) {
+		converter := bindings.NewTypeConverter()
+		converter.SetMode(bindings.REST)
+
+		dataValue, err1 := converter.ConvertToVapi(obj, model.ServiceBindingType())
+		if err1 != nil {
+			return err1[0]
+		}
+		gmObj, err2 := converter.ConvertToGolang(dataValue, gm_model.ServiceBindingType())
+		if err2 != nil {
+			return err2[0]
+		}
+		gmService := gmObj.(gm_model.Service)
+		client := gm_infra.NewDefaultServicesClient(connector)
+		err = client.Patch(id, gmService)
+	} else {
+		client := infra.NewDefaultServicesClient(connector)
+		err = client.Patch(id, obj)
+		if err != nil {
```

Comment:
looks like this should move out of the global manager if?

Reply:
Fixed